src/system/documents/actor.ts

```diff
@@ -565,71 +565,57 @@ export class CosmereActor<
                 ? CONFIG.COSMERE.damageTypes[instance.type]
                 : { ignoreDeflect: false };
 
-            let amount = instance.amount;
-            if (!damageConfig.ignoreDeflect) {
-                // Apply deflect
-                amount = Math.max(0, amount - this.deflect);
-                deflected += this.deflect;
+            let amount = Math.floor(instance.amount);
+            if (!damageConfig.ignoreDeflect && deflected < this.deflect) {
+                // Apply deflect with carry over across multi damage types
+                const remainingDeflect = this.deflect - deflected;
+                const afterDeflect = amount - remainingDeflect;
+                const carryOver = Math.min(0, afterDeflect);
+
+                amount = Math.max(0, afterDeflect);
+
+                deflected += remainingDeflect + carryOver;
             }
 
             if (instance.type === DamageType.Healing) {
-                amount = -amount;
+                amount *= -1;
             }
 
             // Add to running total
             damage += amount;
         });
```

Comment:
Not particularly important point but since we no longer have to apply deflect separately we can probably just sum up all deflectable damage and reduce it by the deflect in one go. Should make this a bit more readable.